### PR TITLE
Allow UNION in matviews

### PR DESCRIPTION
**Pull request by alesapin**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Now it's allowed to `CREATE MATERIALIZED VIEW` with `UNION [ALL]` in query. Behavior is the same as for matview with `JOIN`: **only first table in `SELECT` expression will work as trigger for insert** , all other tables will be ignored. 

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: src/Processors/Transforms/buildPushingToViewsChain.cpp [hunk 1]
```diff
@@ -148,7 +148,7 @@ class ExecutingInnerQueryFromViewTransform final : public ExceptionKeepingTransf
     struct State
     {
         QueryPipeline pipeline;
-        PullingPipelineExecutor executor;
+        PullingAsyncPipelineExecutor executor;
```
Comment (alesapin) [inline src/Processors/Transforms/buildPushingToViewsChain.cpp:151]: Unrelated change, but still makes sense.

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 9c0683e15786727b343d58a6fbd83464f4f9f7e4 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72347/9c0683e15786727b343d58a6fbd83464f4f9f7e4/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/72319/52391a82719f62bd733d9c2b5ca8c7874c14fafb/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Processors/Transforms/buildPushingToViewsChain.cpp (hunk 1 above)
Comment (alesapin) [reply src/Processors/Transforms/buildPushingToViewsChain.cpp]: It breaks some metrics....

**Comment by alesapin** [comment]
`test_concurrent_threads_soft_limit/test.py::test_concurrent_threads_soft_limit_defined_50` - https://github.com/ClickHouse/ClickHouse/issues/71672
`ast fuzzer` -- https://github.com/ClickHouse/ClickHouse/issues/62741
`01290_max_execution_speed_distributed` -- https://github.com/ClickHouse/ClickHouse/issues/71366
``mutations_isolatiuon` --  https://github.com/ClickHouse/ClickHouse/issues/67842

**Comment by clickgapai** [comment]
Hi @alesapin — while reviewing this PR I found the following:

- **Bug (P2):** `checkAllowedQueries` early return bypasses validation of PREWHERE/SAMPLE/FINAL in later UNION branches — https://github.com/ClickHouse/ClickHouse/issues/104768

Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**